sky/utils/controller_utils.py

```diff
@@ -711,7 +711,8 @@ def maybe_translate_local_file_mounts_and_sync_up(task: 'task_lib.Task',
         logger.info(f'{colorama.Fore.YELLOW}Uploading sources to cloud storage.'
                     f'{colorama.Style.RESET_ALL} See: sky storage ls')
     try:
-        task.sync_storage_mounts()
+        # Optimize filemount translation
+        task.compress_local_sync_storage_mounts()
```

Comment:
My code will only apply this to workdirs. At the moment, it seems like it's a good idea to compress regardless of number of files (at least for workdir) as we get a lot of benefit from compression, regardless of small files. One thing we could do is the set it through the client and leave one as default, with three behaviors: compress all, compress if certain number of small files, never compress.

Reply:
I have removed compress_local_sync_storage_mounts() so on the next commit it should be clearer that the compression is workdir only

Reply:
what is the benefit to do compression on small number of small files? I assume it will add the overhead to our system (e.g. compress and decompress)?

Reply:
We still gain benefit if we have a large number of medium sized files, and also just compression in general greatly reduces the number of mbs we upload. So people with lower network bandwidth would benefit from compression a lot more (e g. me). With good network bandwidth or a lot of binary files there is a possibility that there will be higher overhead, but it would be pretty minimal since workdirs aren't meant to be too large anyways. We could also make it a user toggle to reduce this.

Reply:
Lets do some benchmark and get some numbers

Reply:
Turns out that the compression itself isn't very effective at reducing latency, rather the fact that we are sending a single large file holds more weight: Skypilot repo totals to 63M, and compression using tar only puts it down to about 59.5 in this case. However, we see that with compression the total latency is 9s while without it is 26s. This is with 1110 files. Decompression time on GCP VM: 0.126s).
If we were to upload only 8 binary files totalling 177mb, we still see an decrease in time (although less significant). I suspect that this is because we are simply sending less files. The unzip time is not that significant (0.234s).
Compressed: 177.6 MB, (13.7s, 13.4s, 13.35s)
Uncompressed: 178 MB, (15.0s,  15.9s, 14.03s)
When I send a singular binary file, we see no difference in time (as expected).

Conclusion: for compression it makes more sense to simply count total number of files. Since we see an improvement at even 8 files, I think we could just opt for compression of workdir every time or at anything >=10 (some other low number).

Reply:
Thanks! Those numbers looks great. Lets use the >= 10 threshold.

Please also check after sending it into tar, can we still use the incremental upload for rsync.